runtime/compiler/z/codegen/J9AheadOfTimeCompile.cpp

```diff
@@ -280,7 +282,17 @@ uint8_t *J9::Z::AheadOfTimeCompile::initializeAOTRelocationHeader(TR::IteratedEx
          *(uintptrj_t *)cursor = (uintptrj_t) classSymRef->getOwningMethod(self()->comp())->constantPool();
          cursor += SIZEPOINTER;
 
-         *(uintptrj_t *)cursor = (uintptrj_t) classSymRef->getCPIndex(); // cp Index
+         if (comp->getOption(TR_UseSymbolValidationManager))
+            {
+            TR_OpaqueClassBlock *clazz = (TR_OpaqueClassBlock *)recordInfo->data5;
+            uintptrj_t classID = (uintptrj_t)symValManager->getIDFromSymbol(static_cast<void *>(clazz));
+            TR_ASSERT_FATAL(classID, "classID should exist!\n");
```

Comment:
I'm gonna defer to @fjeremic 's thoughts on this one; I just figure it's worth mentioning that though this does mean the AOT compile is compromised, it doesn't necessarily mean the JVM is. Just a `TR_ASSERT` and failing the compilation here might be the better thing to do in a non-debug build.

Reply:
Given that we don't have debug builds where such a `TR_ASSERT` would fire and because this is new code I'd rather us fail completely so we can fix the obvious bug than suffer a silent never to be caught performance issue.